backend/uclapi/roombookings/views.py

```diff
@@ -163,3 +164,66 @@ def get_equipment(request, *args, **kwargs):
         "ok": True,
         "equipment": _serialize_equipment(equipment)
     }, rate_limiting_data=kwargs)
+
+
+@api_view(['GET'])
+@uclapi_protected_endpoint()
+def free_rooms(request, *args, **kwargs):
+    request_params = {}
+    request_params['startdatetime__gte'] = request.GET.get('start_datetime')
+    request_params['finishdatetime__lte'] = request.GET.get('end_datetime')
+
+    if (
+        not request_params['startdatetime__gte'] or
+        not request_params['finishdatetime__lte']
+    ):
+        return PrettyJsonResponse({
+            "ok": False,
+            "error": "start_datetime or end_datetime not provided"
+        })
```

Comment:
The rate limit will still have been applied so we should pass `kwargs` to the `PrettyJsonResponse` too so that the headers get sent to the client.